stairway/src/main/java/bio/terra/stairway/impl/FlightFilterAccess.java

```diff
@@ -226,17 +200,19 @@ private void makeFlightFilter(StringBuilder sb, String prefix) {
     }
   }
 
-  /**
-   * Generate the filter for the flight input table for Form 3: the case where there is more that
-   * one predicate.
-   */
-  private void makeInputFilter(StringBuilder sb) {
-    String inter = StringUtils.EMPTY;
-
-    for (FlightFilterPredicate predicate : filter.getInputPredicates()) {
-      String sql = makeInputPredicateSql(predicate);
-      sb.append(inter).append(sql);
-      inter = " OR ";
+  // -- boolean expression methods
+  private String makeBooleanExpressionsFilters(FlightBooleanOperationExpression expression) {
+    switch (expression.getOperation()) {
+      case PREDICATE:
+        return makeInputPredicateSql(expression.getBasePredicate());
+      case AND, OR:
+        return "("
+            + expression.getExpressions().stream()
+                .map(this::makeBooleanExpressionsFilters)
+                .collect(Collectors.joining(expression.getOperation().getSql()))
+            + ")";
```

Comment:
You can simplify this by using the `joining()` form that supports prefix and suffix.
```suggestion
        return expression.getExpressions().stream()
                .map(this::makeBooleanExpressionsFilters)
                .collect(Collectors.joining(expression.getOperation().getSql(), "(", ")"));
```

Reply:
Oh, interesting...I didn't realize you could prefix and suffix there too.  cool